### ISSUE TITLE
Set tracking mode (MGLUserTrackingModeFollowWithHeading) issue

### ISSUE THREAD

**Issue by MattCarr147** [mapbox/mapbox-gl-native#13772]
The following code does not function correctly if the user location annotation is not already visible.

```
mapView.showsUserLocation = true
mapView.setUserTrackingMode(.followWithHeading, animated: true)
```

I've encountered this whilst implementing a UI which allows the user to select from the various tracking modes in addition to a 'none' state (MGLUserTrackingMode.none & showsUserLocation = false).

When I run the code from the 'none' state, the map displays the user location but does nothing else. 

My workaround at the moment is to show the user location and then after a brief delay (0.1 seconds), set tracking to .followWithHeading. This seems to work. 

Note that if I go from the 'none' state to ```MGLUserTrackingMode.follow``` everything works fine without needing the delay. 

### Steps to reproduce

 1. Load a map with showsUserLocation = false and trackingMode = .none
 2. Trigger showsUserLocation = true and trackingMode = .followWithHeading via a user interaction
 3. The user location appears on the map but nothing else happens. No heading indicator is displayed and no tracking behaviour occurs. 

### Expected behaviour

The user's location should display on the map and be centered, as should the visual 'heading' indicator. The map should track the location and automatically pan as the device is moved around. 

### Actual behaviour

User location is shown on the map but is not centered. The heading indicator isn't displayed. No tracking of position occurs.

### Configuration

**Mapbox SDK versions:** 4.7.1
**iOS/macOS versions:** iOS 12.0.1
**Device/simulator models:** iPad Mini 4th Generation
**Xcode version:** 10.0

**Comment by julianrex** [comment]
@MattCarr147 thanks for the bug report! Do you have a sample app that demonstrates this issue?

**Comment by MattCarr147** [comment]
Hey, no problem at all. 

Sorry, but I don't have a project I can share at the moment as I encountered the issue when working on proprietary code. I'll look into putting a sample project together when I get a chance, but I can't make doing so a priority unfortunately.

**Comment by fabian-guerra** [comment]
Fixed in #13849
